homeassistant/components/knx/strings.json

```diff
@@ -0,0 +1,42 @@
+{
+  "config": {
+    "step": {
+      "type": {
+        "description": "Please enter the connection type we should use for your KNX connection. \n AUTOMATIC - The integration takes care of the connectivity to your KNX Bus by performing a gateway scan. \n TUNNELING - The integration will connect to your KNX bus via tunneling. \n ROUTING - The integration will connect to your KNX bus via routing.",
+        "data": {
+          "connection_type": "KNX Connection Type"
+        }
+      },
+      "tunnel": {
+        "description": "Please select a gateway from the list.",
+        "data": {
+          "gateway": "KNX Tunnel Connection"
+        }
+      },
+      "manual_tunnel": {
+        "description": "Please enter the connection information of your tunneling device.",
+        "data": {
+          "port": "[%key:common::config_flow::data::port%]",
+          "host": "[%key:common::config_flow::data::host%]",
+          "individual_address": "Individual address for the connection",
+          "route_back": "When True the KNXnet/IP Server shall use the IP address and the port number from the IP package received as the target IP address or port number for the response to the KNXnet/IP Client (for NAT/Docker)"
+        }
+      },
+      "routing": {
+        "description": "Please configure the routing options.",
```

Comment:
Does this need to be configured as part of the initial config flow or could this be either moved to YAML or to the options flow ? Or maybe only shown for advanced users?

Reply:
Routing is an alternative connection mode to Tunnelling (Multicast vs Unicast). It's either the one or the other, both need different connection parameters set.

Reply:
I'd leave it in there as the goal is to get rid of the connection specific settings in the YAML and only have the different platform configs in our YAML. (We will need to add an options flow for the remaining options)